src/main/java/org/isf/opd/service/OpdIoOperationRepository.java

```diff
@@ -54,5 +58,18 @@ public interface OpdIoOperationRepository extends JpaRepository<Opd, Integer>, O
 	@Query(value = "select op from Opd op where op.prog_year = :prog_year and op.date >= :dateVisitFrom and op.date < :dateVisitTo")
 	List<Opd> findByProgYearAndDateBetween(@Param("prog_year") Integer prog_year, @Param("dateVisitFrom") LocalDateTime dateVisitFrom,
 			@Param("dateVisitTo") LocalDateTime dateVisitTo);
+	
+	@Query("select o from Opd o order by o.prog_year")
+	Page<Opd> findAllOrderByProgYearDescPageable(Pageable pageable);
+	
+	@Query("select o from Opd o where o.patient.code = :code order by o.prog_year")
+	Page<Opd> findAllByPatient_CodeOrderByProgYearDescPageable(@Param("code") Integer code, Pageable pageable);
+	
+	@Query(value = "select op from Opd op where op.ward = :ward or op.disease.diseaseType = :diseaseType or op.disease.code = :diseaseCode or (op.date >= :dateFrom and op.date < :dateTo) "
+			+ " or (op.age >= :ageFrom and op.age < :ageTo) or op.sex = :sex or op.newPatient = :newPatient")
+	Page<Opd> findOpdListPageable(@Param("ward") Ward ward, @Param("diseaseType") DiseaseType diseaseType, @Param("diseaseCode") String diseaseCode, 
```

Comment:
why all equations in `or` relation?

```suggestion
	@Query(value = "select op from Opd op where op.ward = :ward and op.disease.diseaseType = :diseaseType and op.disease.code = :diseaseCode and (op.date >= :dateFrom and op.date < :dateTo) and (op.age >= :ageFrom and op.age < :ageTo) and op.sex = :sex and op.newPatient = :newPatient")
	Page<Opd> findOpdListPageable(@Param("ward") Ward ward, @Param("diseaseType") DiseaseType diseaseType, @Param("diseaseCode") String diseaseCode, 
```